### ISSUE TITLE
Error: EISDIR: illegal operation on a directory, read when running rollup -c

### ISSUE THREAD

**Issue by amiregelz** [extend-chrome/rollup-plugin-chrome-extension#186]
### Build tool

Rollup

### Where do you see the problem?

- [ ] In the browser
- [X] In the terminal

### Describe the bug

Trying RPCE for the first time and using the simple starter snippet from the Usage section.
When I run `rollup -c` I am getting this cryptic error:
```
> rollup -c --validate

manifest.json → build...
[!] (plugin chrome-extension) Error: EISDIR: illegal operation on a directory, read
Error: EISDIR: illegal operation on a directory, read
```

I tried both on Windows and on Linux (mac).

This is my `rollup.config.js`:
```
import resolve from '@rollup/plugin-node-resolve'
import commonjs from '@rollup/plugin-commonjs'

import { chromeExtension, simpleReloader } from 'rollup-plugin-chrome-extension'

export default {
    input: 'manifest.json',
    output: {
        dir: 'build',
        format: 'esm',
    },
    plugins: [
        // always put chromeExtension() before other plugins
        chromeExtension(),
        simpleReloader(),
        // the plugins below are optional
        resolve(),
        commonjs(),
    ],
}
```

My `manifest.json` is standard as well:
```
{
  "manifest_version": 2,
  "permissions": [
    ...
  ],
  "background": {
    "scripts": [
      "background_script/background.js"
    ]
  },
  "content_scripts": [
    {
      "matches": [
        "https://website.com/*"
      ],
      "js": [
        "content_script/content.js"
      ],
      "css": [
        "content_script/css/style.css"
      ]
    }
  ],
  "browser_action": {
    "default_title": "title",
    "default_icon": {
      "16": "static/icons/16.png",
      "32": "static/icons/32.png"
    }
  },
  "icons": {
    "16": "static/icons/16.png",
    "32": "static/icons/32.png",
    "48": "static/icons/48.png",
    "128": "static/icons/128.png"
  },
  "web_accessible_resources": [
    "static/*"
  ]
}
```

What should I investigate?

### Reproduction

Just the basic usage example. my repo is private but is there anything I should investigate?

### System Info

```shell
System:
    OS: Windows 10 10.0.17763
    CPU: (4) x64 Intel(R) Core(TM) i3-7100 CPU @ 3.90GHz
    Memory: 1.01 GB / 7.89 GB
  Binaries:
    Node: 14.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.6.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.11 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 97.0.4692.71
    Edge: Spartan (44.17763.831.0)
    Internet Explorer: 11.0.17763.771
```

### Severity

blocking all usage of RPCE

**Comment by jacksteamdev** [comment]
Hard to say without more info. The manifest looks good. It seems that Rollup is trying to read a directory as a file.

What version of RPCE are you using?

**Comment by amiregelz** [comment]
The latest, tried both beta and stable

**Comment by jacksteamdev** [comment]
@amiregelz I've done a major rewrite, would you mind giving it a try?

The API has changed a little, you can see the updated instructions on DEV: 

[Create a Vite-React Chrome Extension in 90 seconds - DEV Community 👩‍💻👨‍💻](https://dev.to/jacksteamdev/create-a-vite-react-chrome-extension-in-90-seconds-3df7)

**Comment by flurrux** [comment]
i had the same problem as [amiregelz](https://github.com/amiregelz) and after following your vite-react tutorial it now works.  
it didn't work right away though, only after these two changes:  
- [migrate from manifest version 2 to 3](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/)
with manifest version 2, i got this error: 
![Screenshot 2022-03-16 235358](https://user-images.githubusercontent.com/40806446/158826510-0e112611-64bb-4ebf-9832-f990ed61a371.jpg)
which i think occurs at [this point in the code](https://github.com/extend-chrome/rollup-plugin-chrome-extension/blob/0c0ce568d2e5ec08083e540cdf7088f89db4b795/src/node/plugin-resources.ts#L214) because `web_accessible_resources` is assumed to be a list of objects rather than a list of strings as per [MV3](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#web-accessible-resources).  
- moving all of my icons to the toplevel `./public` folder as suggested [here](https://github.com/extend-chrome/rollup-plugin-chrome-extension/issues/226#issuecomment-1059220327)  

now it works like a charm!  

@jacksteamdev can i also just say that your blog post and comments are such a joy to read. i find them very comprehensible and user friendly. thank you for the effort you put into it all :)

**Comment by jacksteamdev** [comment]
@flurrux Thanks for your kind words! Comments like yours make all the difference. :hugs: 

We probably need to add some validation, since manifest version 2 isn't supported.

**Comment by jacksteamdev** [comment]
Added MV3 validation in #258

**Comment by jacksteamdev** [comment]
> moving all of my icons to the top-level `./public` folder as suggested

Added auto-copy for missing manifest assets in #259

**Comment by jacksteamdev** [comment]
It appears that this issue is now primarily a duplicate of #137, closing
